libflux/src/flux/semantic/builtins.rs

```diff
@@ -275,6 +275,267 @@ pub fn builtins() -> Builtins<'static> {
                 "splitRegexp" => Node::Builtin("forall [] (r: regexp, v: string, i: int) -> [string]"),
                 "getString" => Node::Builtin("forall [] (r: regexp) -> string"),
             }),
+            "universe" => Node::Package(maplit::hashmap! {
+                "inf" => Node::Builtin("forall [] duration"),
+                "int" => Node::Builtin("forall [t0] (v: t0) -> int"),
+                "integral" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?unit: duration,
+                        ?timeColumn: string,
+                        ?column: string
+                    ) -> [t1]
+                "#),
+                "join" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: t0,
+                        ?method: string,
+                        ?on: [string]
+                    ) -> [t1]
+                "#),
+                // This function would almost have input/output types that match, but:
+                // input column may start as int, uint or float, and always ends up as float.
+                "kaufmansAMA" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        n: int,
+                        ?column: string
+                    ) -> [t1]
+                "#),
+                // either column list or predicate must be provided
+                "keep" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?columns: [string],
+                        ?fn: (column: string) -> bool
+                    ) -> [t1]
+                "#),
+                // the "fn" parameter to keyValues is suspicous: https://github.com/influxdata/flux/issues/2235
+                "keyValues" => Node::Builtin(r#"
+                    forall [t0, t1, t2, t3] where t0: Row, t3: Row (
+                        <-tables: [t0],
+                        ?keyColumns: [string],
+                        ?fn: t1
+                    ) -> [{_key: string | _value: t2 | t3}]
+                "#),
```

Comment:
Per the issue you linked, can we just remove `fn` from the type signature?

Reply:
Sure, I'll just remove it here, and save removing it from the Go code to a separate PR linked to that issue.